src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -173,4 +174,16 @@ public static long scaleOffset(MemorySegment segment, long index, long size) {
         // note: we know size is a small value (as it comes from ValueLayout::byteSize())
         return MemorySegmentProxy.multiplyOffsets(index, (int)size, (AbstractMemorySegmentImpl)segment);
     }
+
+    @Stable
+    public static LayoutAccess LAYOUT_ACCESS;
```

Comment:
Clever :) Maybe some comment here about what this is doing would be nice (I had to look at it for a while).
```suggestion
    public static LayoutAccess LAYOUT_ACCESS; // shared secret
```

Alternatively, I think you could do this with a MethodHandle acquired through `Lookup.privateLookupIn(lookup(), ValueLayout.class).findVirtual(...)`.